tests/testthat/test-scalevariableset.R

```diff
@@ -208,40 +208,99 @@ test_that("PickOneMulti with merge, hide, NET",
+    "C", "B", "A")), class = "data.frame", names = "", row.names = c(NA,
+    6L))), values = c(A = 1, B = 2, C = 3, B = 4, A = 5), sourcevalues = c(A = 1,
+B = 2, C = 3, B = 4, A = 5), variablevalues = list(c(A = 1, B = 2,
+C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1,
 B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5
-), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, 
-B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, 
+), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3,
+B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1,
 B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5
-), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, 
+), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3,
 B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5)), variablesourcevalues = list(
-    c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, 
-    B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, 
-    B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, 
-    A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, 
+    c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3,
+    B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1,
+    B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4,
+    A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2,
     C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5
-    ), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2, 
+    ), c(A = 1, B = 2, C = 3, B = 4, A = 5), c(A = 1, B = 2,
     C = 3, B = 4, A = 5), c(A = 1, B = 2, C = 3, B = 4, A = 5
     ), c(A = 1, B = 2, C = 3, B = 4, A = 5)), codeframe = list(
     A = 1, B = 2, C = 3, B = 4, A = 5, NET = c(1, 2, 3, 4, 5)), secondarycodeframe = list(
-    `Label 1` = 0L, `Label 2` = 1L, `Label 3` = 2L, `Label 4` = 3L, 
-    `Label 5` = 4L, `Label 6` = 5L, `Label 7` = 6L, `Label 8` = 7L, 
+    `Label 1` = 0L, `Label 2` = 1L, `Label 3` = 2L, `Label 4` = 3L,
+    `Label 5` = 4L, `Label 6` = 5L, `Label 7` = 6L, `Label 8` = 7L,
     `Label 9` = 8L, `Label 10` = 9L, `Label 11` = 10L, `Label 12` = 11L), transposed = TRUE, class = "data.frame")
 
     expect_error(ScaleVariableSet(test.case, type = 'standardize', within.case = TRUE))
 })
+
+test_that("DS-3901: Ensure ScaleVariableSet handles Dates and single variables", {
+    file <- system.file("extdata", "ds3901testcases.rds", package = "flipTransformations")
+    test.cases <- readRDS(file)
+
+    # Date
+    std.res <- ScaleVariableSet(test.cases[[1]], type = "standardize", within.case = FALSE)
+    expect_equal(mean(std.res, na.rm = TRUE), 0, tolerance = 1e-7)
+    expect_equal(sd(std.res, na.rm = TRUE), 1, tolerance = 1e-7)
+
+    center.res <- ScaleVariableSet(test.cases[[1]], type = "center", within.case = FALSE)
+    expect_equal(mean(center.res, na.rm = TRUE), 0, tolerance = 1e-7)
+
```

Comment:
Could update the test case `rds` file to be a named list? Might make the tests easier to understand?
e.g.

```
unit.res <- ScaleVariableSet(test.cases[["Date"]], type = "unit", within.case = FALSE)
    expect_equal(min(unit.res, na.rm = TRUE), 0, tolerance = 1e-7)
    expect_equal(max(unit.res, na.rm = TRUE), 1, tolerance = 1e-7)
```